Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -0,0 +1,712 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Sycl/Seeding/Seedfinder.hpp"
+#include "Acts/Plugins/Sycl/Seeding/detail/Types.h"
+
+#include <algorithm>
+#include <array>
+#include <atomic>
+#include <cstdint>
+#include <cstring>
+#include <exception>
+#include <iterator>
+#include <numeric>
+#include <vector>
+
+#include <CL/sycl.hpp>
+
+namespace Acts::Sycl {
+/// Kernel classes in order of execution.
+class duplet_search_bottom_kernel;
+class duplet_search_top_kernel;
+class ind_copy_bottom_kernel;
+class ind_copy_top_kernel;
+class transform_coord_bottom_kernel;
+class transform_coord_top_kernel;
+class triplet_search_kernel;
+class filter_2sp_fixed_kernel;
+
+// Returns the smallest multiple of workGroupSize that is not smaller
+// than numThreads.
+uint32_t numWorkItems(uint32_t numThreads, uint32_t workGroupSize) {
+  auto q = numThreads / workGroupSize + (numThreads % workGroupSize != 0);
```

Comment:
You can achieve the same with:

```c++
(numThreads + workGroupSize - 1) / workGroupSize
```

Which avoids using the `%` operator. (Which can be rather slow.)